Traceback (most recent call last):
  File "/home/sa7270/diffusion/run_experiment.py", line 44, in <module>
    diffusion.run()
  File "/home/sa7270/diffusion/diffusion.py", line 54, in run
    self.generate_data()
  File "/home/sa7270/diffusion/diffusion.py", line 41, in generate_data
    self.X_train.gen_x1_xt()
  File "/home/sa7270/diffusion/data.py", line 47, in gen_x1_xt
    x_t = self.ξ * self.α(self.t) + x_1 * self.β(self.t)
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 4.77 GiB. GPU 0 has a total capacity of 31.74 GiB of which 2.72 GiB is free. Including non-PyTorch memory, this process has 29.01 GiB memory in use. Of the allocated memory 28.63 GiB is allocated by PyTorch, and 12.37 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
